### PR TITLE
Functional tests : BO - My Profile

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | Functional tests : BO - My Profile
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| Related PRs       | This PR #28985 will update an error message.
| How to test?      | Run `HEADLESS=false TEST_PATH="functional/BO/15*/07*" URL_FO="https://domain.tld" npm run test:specific`
| Possible impacts? | N/A

### CODE REVIEW

File: tests/UI/commonTests/BO/advancedParameters/setPermissions.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      await testContext.addContextItem(this, 'testIdentifier', 'goToProfileSubTab', baseContext);
+
+      const isSubTabOpened = await permissionsPage.goToProfileSubTab(page, profileName);
+      await expect(isSubTabOpened, 'Profile sub-tab is not opened!').to.be.true;
+    });
+
+    permissionsData.forEach((permission) => {
+      permission.accesses.forEach((access) => {
+        it(`should set the permission ${access} on the ${permission.className}`, async () => {
+          const isPermissionDefined = await permissionsPage.setPermission(page, permission.className, access);
```
Comment (nesrineabdmouleh) [inline tests/UI/commonTests/BO/advancedParameters/setPermissions.js:81]:
```suggestion
          await testContext.addContextItem(this, 'testIdentifier', `setPermission${permission.className}`, baseContext);

          const isPermissionDefined = await permissionsPage.setPermission(page, permission.className, access);
```

File: tests/UI/commonTests/BO/advancedParameters/setPermissions.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+let browserContext;
+let page;
+
+/**
+ *
+ * @param {string} profileName
+ * @param {array} permissionsData
+ * @param {string} baseContext
+ */
```
Comment (nesrineabdmouleh) [inline tests/UI/commonTests/BO/advancedParameters/setPermissions.js]:
```suggestion
/**
 * Function to set the employee permissions
 * @param profileName {string} 
 * @param permissionsData {array} 
 * @param baseContext {string} 
 */
```
Please add the description of all params

File: tests/UI/pages/BO/advancedParameters/team/permissions/index.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+/**
+ * Permissions page, contains functions that can be used on the page
+ * @class
+ * @extends BOBasePage
+ */
+class Permissions extends BOBasePage {
+  /**
+   * @constructs
+   * Setting up texts and selectors to use on profiles page
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/advancedParameters/team/permissions/index.js]:
```suggestion
   * Setting up texts and selectors to use on permissions page
```

File: tests/UI/pages/BO/advancedParameters/team/permissions/index.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    // Selectors
+    this.growlMessageBlock = `${this.growlDiv} .growl-message`;
+    this.growlCloseButton = `${this.growlDiv} .growl-close`;
+    this.successfulUpdateMessage = 'Update successful';
+    this.profileSubTab = profileName => `a[id^="profile-"]:text("${profileName}")`;
+    this.profileAccess = (className, access) => `input[data-type="${access}"][data-classname="${className}"]:visible`;
+
+    // List of profiles
+
+    // List of permissions
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/advancedParameters/team/permissions/index.js]: Maybe delete these comments

File: tests/UI/pages/BO/advancedParameters/team/employeeBasePage.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+/**
+ * Employee base page, contains functions that can be used on the page
+ * @class
+ * @extends BOBasePage
+ */
+module.exports = class EmployeeBasePage extends BOBasePage {
+  /**
+   * @constructs
+   * Setting up texts and selectors to use on localization page
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/advancedParameters/team/employeeBasePage.js]:
```suggestion
   * Setting up texts and selectors to use on Employee page
```

File: tests/UI/pages/BO/advancedParameters/team/permissions/index.js (hunk 3 above)
Comment (Progi1984) [reply tests/UI/pages/BO/advancedParameters/team/permissions/index.js]: Fixed

File: tests/UI/commonTests/BO/advancedParameters/setPermissions.js (hunk 2 above)
Comment (Progi1984) [reply tests/UI/commonTests/BO/advancedParameters/setPermissions.js]: Fixed

File: tests/UI/pages/BO/advancedParameters/team/employeeBasePage.js (hunk 5 above)
Comment (Progi1984) [reply tests/UI/pages/BO/advancedParameters/team/employeeBasePage.js]: Fixed

File: tests/UI/pages/BO/advancedParameters/team/permissions/index.js (hunk 4 above)
Comment (Progi1984) [reply tests/UI/pages/BO/advancedParameters/team/permissions/index.js]: Fixed

Comment (Progi1984) [reply tests/UI/pages/BO/advancedParameters/team/permissions/index.js]: Fixed

**Review by nesrineabdmouleh — DISMISSED** [review]

**Comment by Progi1984** [comment]
@nesrineabdmouleh Just rebased (like asked by @boubkerbribri)

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Pull request merged**